measure_optimize.py

```diff
@@ -2,180 +2,2412 @@
+        None
         """
-        # check if measurement vector legal
-        assert len(measurement_vector)==self.total_no_measure, "Measurement vector is of wrong shape!!!"
+        if self.fixed_nlp:
+            solver = pyo.SolverFactory("cyipopt")
+            solver.config.options["hessian_approximation"] = "limited-memory"
+            additional_options = {
+                "max_iter": 3000,
+                "output_file": "console_output",
+                "linear_solver": "mumps",
+                # "halt_on_ampl_error": "yes", # this option seems not working for cyipopt
+                "bound_push": 1e-10,
+            }
 
-        measurement_matrix = np.zeros((self.total_no_measure, self.total_no_measure))
+            if degeneracy_hunter:
+                additional_options = {
+                    "max_iter": 0,
+                    "output_file": "console_output",
+                    "linear_solver": "mumps",
+                    "bound_push": 1e-10,
+                }
 
-        for i in range(self.total_no_measure):
-            for j in range(self.total_no_measure):
-                measurement_matrix[i,j] = min(measurement_vector[i], measurement_vector[j])
+            # copy solver options
+            for k, v in additional_options.items():
+                solver.config.options[k] = v
+            results = solver.solve(self.mod, tee=True)
 
-        return measurement_matrix
+        elif not mip_option and objective == ObjectiveLib.A:
+            # solver = pyo.SolverFactory('ipopt')
+            # solver.options['linear_solver'] = "ma57"
+            # results = solver.solve(self.mod, tee=True)
+
+            solver = pyo.SolverFactory("gurobi", solver_io="python")
+            solver.options["mipgap"] = 0.1
+            results = solver.solve(self.mod, tee=True)
 
-    def __print_FIM(self, FIM):
+        elif mip_option and objective == ObjectiveLib.A:
+            solver = pyo.SolverFactory("gurobi", solver_io="python")
+            # solver.options['mipgap'] = 0.1
+            results = solver.solve(self.mod, tee=True)
+
+        elif not mip_option and objective == ObjectiveLib.D:
+            solver = pyo.SolverFactory("cyipopt")
+            solver.config.options["hessian_approximation"] = "limited-memory"
+            additional_options = {
+                "max_iter": 3000,
+                "output_file": "console_output",
+                "linear_solver": "mumps",
+            }
+
+            if degeneracy_hunter:
+                additional_options = {
+                    "max_iter": 0,
+                    "output_file": "console_output",
+                    "linear_solver": "mumps",
+                    "bound_push": 1e-6,
+                }
+
+            # copy solver options
+            for k, v in additional_options.items():
+                solver.config.options[k] = v
+            results = solver.solve(self.mod, tee=True)
+
+        elif mip_option and objective == ObjectiveLib.D:
```

Comment:
Should one of the objective options be mixed?

Reply:
yes.